pkg/api/api.go

```diff
@@ -198,7 +198,7 @@ func (hs *HttpServer) registerRoutes() {
 			r.Get("/", GetDataSources)
 			r.Post("/", quota("data_source"), bind(m.AddDataSourceCommand{}), AddDataSource)
 			r.Put("/:id", bind(m.UpdateDataSourceCommand{}), wrap(UpdateDataSource))
-			r.Delete("/:id", DeleteDataSource)
+			r.Delete("/:idOrName", DeleteDataSource)
```

Comment:
I think this url should follow the same pattern as `r.Get("/name/:name")`. Using the same handler for both name and id makes it impossible to use a number as name.